go/api/go_api/tests/utils.py

```diff
@@ -1,5 +1,27 @@
 from mock import Mock, patch
 
+from vumi.tests.utils import MockHttpServer
```

Comment:
Practical question -- how would we make `treq` access the fake agent provided by `FakeHttpServer.get_agent()`? I'm sure it's possible, but not sure it's possible without making a bit of a mess.

Philosophical disagreement -- if a TCP packet randomly fails to traverse the localhost network, your system is very broken. I have yet to see it happen. Spinning up a temporary HTTP server is simple and provides a clean API that matches how the code is used outside of tests. `FakeHttpServer`on the other hand is 300 lines of overriding Twisted agents and protocols, which doesn't fill me with confidence.